### PR TITLE
Campaign, some names of bonuses don't fit the box and should be shortened #3509

### LINKED ISSUE #3509: Campaign, some names of bonuses don't fit the box and should be shortened
The names of bonuses in "Choice" box are sometimes too long:
image
image
image
image
image
image
image
image
image

### PR DESCRIPTION
**Pull request by felix642**
Fix #3509

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @felix642 , please fix code styling issues highlighted by **Code style check / clang-format (pull_request)** runner.

File: src/fheroes2/resource/artifact.cpp [hunk 1]
```diff
@@ -313,6 +314,30 @@ const char * Artifact::GetName( void ) const
     return _( artifacts[id].name );
 }
 
+const char * Artifact::GetCampaignName( void ) const
```
Comment (ihhub) [inline src/fheroes2/resource/artifact.cpp:317]: Please put this function into anonymous namespace in `campaign_scenariodata.cpp` file as this is campaign specific information only.

File: src/fheroes2/spell/spell.cpp [hunk 2]
```diff
@@ -228,6 +228,16 @@ const char * Spell::GetName( void ) const
     return _( spells[id].name );
 }
 
+const char * Spell::GetCampaignName( void ) const
```
Comment (ihhub) [inline src/fheroes2/spell/spell.cpp:231]: Please put this function into anonymous namespace in `campaign_scenariodata.cpp` file as this is campaign specific information only.

File: src/fheroes2/resource/artifact.cpp (hunk 1 above)
Comment (ihhub) [inline src/fheroes2/resource/artifact.cpp:317]: Also please name the function from **non** capital letter. Once this function is moved to the described file it should accept `const int32_t artifactId` argument.

File: src/fheroes2/resource/artifact.cpp [hunk 3]
```diff
@@ -62,116 +62,117 @@ struct artifactstats_t
 };
 
 artifactstats_t artifacts[] = {
-    {0, 12, TYPE3, _( "Ultimate Book of Knowledge" ), _( "The %{name} increases your knowledge by %{count}." )},
```
Comment (ihhub) [inline src/fheroes2/resource/artifact.cpp:65]: After required changes this file will no longer needed to be updated. Let's keep please changes minimal for now.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @felix642 , I left few comments here. Would you mind to take a look?

File: src/fheroes2/resource/artifact.cpp (hunk 1 above)
Comment (felix642) [reply src/fheroes2/resource/artifact.cpp:317]: I wasn't  quite sure to understand the coding style. Some functions where named using upper camel case while others were using lower camel case. Is there a specific rule for that ?

File: src/fheroes2/resource/artifact.cpp (hunk 3 above)
Comment (felix642) [reply src/fheroes2/resource/artifact.cpp:65]: Yes, I'm having a little bit of trouble with clang-format. I'll take a look at that later!

Comment (felix642) [reply src/fheroes2/resource/artifact.cpp:65]: Yes, I'm having a bit of trouble with clang-format I'll take a look at that later !

File: src/fheroes2/resource/artifact.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/resource/artifact.cpp]: My apologies for this. We inherited a a lot of old code from previous team and the coding style there wasn't present. Per our standard a function should be named with non-capital letter.

File: src/fheroes2/resource/artifact.cpp (hunk 3 above)
Comment (ihhub) [reply src/fheroes2/resource/artifact.cpp]: I'm sorry about this again. At the start of the project we formatted everything but then CI server got an updated version of clang-format (even with the same rules) leading to such situations.

**Comment by ihhub** [comment]
@felix642 , a bunch of thanks for fixing this issue!

**Pull request merged**